torch_geometric_temporal/nn/attention/2sagcn.py

```diff
@@ -0,0 +1,307 @@
+
+        if self.attention:
+            # spatial attention
+            se = y.mean(-2)  # N C V
+            se1 = self.sigmoid(self.conv_sa(se))
+            y = y * se1.unsqueeze(-2) + y
+
+            # temporal attention
+            se = y.mean(-1)
+            se1 = self.sigmoid(self.conv_ta(se))
+            y = y * se1.unsqueeze(-1) + y
+
+            # channel attention
+            se = y.mean(-1).mean(-1)
+            se1 = self.relu(self.fc1c(se))
+            se2 = self.sigmoid(self.fc2c(se1))
+            y = y * se2.unsqueeze(-1).unsqueeze(-1) + y
+
+        return y
+
+
+
+class TCN_GCN_unit(nn.Module):
+    r"""Two-Stream Adaptive Graph Convolutional Network.
+
+    For details see this paper: `"Two-Stream Adaptive Graph Convolutional Networks for Skeleton-Based Action Recognition." <https://arxiv.org/abs/1805.07694>`_
+    This implementation is based on the authors Github Repo https://github.com/lshiwjx/2s-AGCN.
+    It's used by the author for classifying actions from sequences of 3D body joint coordinates. 
+
+    Args:
+        in_channels (int): Number of input features.
+        out_channels (int): Number of output features.
+        edge_index (PyTorch LongTensor): Graph edge indices.
+        num_nodes (int): Number of nodes in the network.
+        stride (int, optional): Time strides during temporal convolution.
+        residual (bool, optional): Applying connections.
+        adaptive (bool, optional): Applying adaptive graphs
+        attention (bool, optional): Applying Attention
+    """
+    def __init__(self, in_channels:int, out_channels:int, edge_index:torch.LongTensor, num_nodes:int, stride:int=1, residual:bool=True, adaptive:bool=True, attention:bool=True):
+        super(TCN_GCN_unit, self).__init__()
+        self.edge_index = edge_index
+        self.num_nodes = num_nodes
+
+        self.graph = Graph(self.edge_index, self.num_nodes)
+        self.A = self.graph.A
+
+
+        self.gcn1 = unit_gcn(in_channels, out_channels, self.A, adaptive=adaptive, attention=attention)
+        self.tcn1 = unit_tcn(out_channels, out_channels, stride=stride)
+        self.relu = nn.ReLU(inplace=True)
+        self.attention = attention
+
+        if not residual:
+            self.residual = lambda x: 0
+
+        elif (in_channels == out_channels) and (stride == 1):
+            self.residual = lambda x: x
+
+        else:
+            self.residual = unit_tcn(in_channels, out_channels, kernel_size=1, stride=stride)
+
+    def forward(self, x):
+        """
+        Making a forward pass.
+        
+        Arg types:
+            * **X** (PyTorch FloatTensor) - Node features for T time periods, with shape (B, F_in, T_in, N_nodes).
+
+        Return types:
+            * **X** (PyTorch FloatTensor)* - Sequence of node features, with shape (B, out_channels, T_in//stride, N_nodes).
+        """
+
+        y = self.relu(self.tcn1(self.gcn1(x)) + self.residual(x))
+
+        return y
+
+
+if __name__ == "__main__":
```

Comment:
We write separate test for doing some sanity checks in the test folder. Test method names start with test. This way we can check the coverage of the test.